File: src/main/java/javabot/JavabotThreadFactory.java
package javabot;

import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

public class JavabotThreadFactory implements ThreadFactory {
    private final AtomicInteger threadNumber = new AtomicInteger(1);
    private final String namePrefix;
    private final boolean createDaemonThreads;

    public JavabotThreadFactory(final boolean daemon, final String name) {
        createDaemonThreads = daemon;
        namePrefix = name;
    }

    public Thread newThread(final Runnable runnable) {
        final Thread t = new Thread(runnable, namePrefix + threadNumber.getAndIncrement());
        t.setDaemon(createDaemonThreads);
        if (t.getPriority() != Thread.NORM_PRIORITY) {
            t.setPriority(Thread.NORM_PRIORITY);
        }
        return t;
    }
}

File: src/main/kotlin/javabot/JavabotThreadFactory.kt
package javabot

import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

class JavabotThreadFactory(
    private val createDaemonThreads: Boolean,
    private val namePrefix: String,
) : ThreadFactory {
    private val threadNumber = AtomicInteger(1)

    override fun newThread(runnable: Runnable): Thread {
        val t = Thread(runnable, namePrefix + threadNumber.andIncrement)
        t.isDaemon = createDaemonThreads
        if (t.priority != Thread.NORM_PRIORITY) {
            t.priority = Thread.NORM_PRIORITY
        }
        return t
    }
}
